### PR TITLE
Feature/UI 65 textarea component

### PR DESCRIPTION
**Pull request by PiotrGrzechnik**
#### Task

- **Issue or task referred:** UI-65
- **Feature or Bugfix:** Feature

It's textarea component which allow to pass as props another component (with autosize behavior). Also user can block manual resizing, sets max number of characters, define custom placeholder, and disable textarea .

#### Checklist

- [x] Required unit tests prepared
- [x] Documentation prepared
- [x] External licenses validated (ideally MIT)

### CODE REVIEW

File: packages/textarea/README.md [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+## Supported props
+
+Property name    | Type      | Default | Description                    
+-----------------|-----------|:-------:|--------------------------------
+className        | string    | n/a     | Additional class name passed to wrapper
+disabled         | bool      | n/a     | Read-only textarea
+maxLength        | number    | n/a     | Define max number of characters in texarea
+placeholder      | string    | n/a     | Custom placeholder to show in textarea
+resize           | bool      | `true`  | Allow to resize textarea
+TextareaComponent| string or node  | `'textarea'`  | Allow to use `'textarea'` tag or TextareaAutosize component
```
Comment (maciek15) [inline packages/textarea/README.md]: You could indent previous lines to match this one.

File: packages/textarea/src/Textarea.js [hunk 2]
```diff
@@ -0,0 +1,59 @@
...
+ * @param {bool} [props.resize]
+ * @param {string|node} [props.TextareaComponent]
+ * @returns {React.Element}
+ */
+function Textarea (props) {
+  const { className, disabled, maxLength, placeholder, resize, TextareaComponent, ...passedProps } = props
+
+  return (
+    <TextareaComponent
+      className={buildClassName('textarea', className, [resize ? null : 'no-resize', disabled ? 'disabled' : null])}
```
Comment (maciek15) [inline packages/textarea/src/Textarea.js]: `[resize ? null : 'no-resize', disabled ? 'disabled' : null]`
can be changed to:
`{ 'no-resize': resize, disabled }`

File: packages/textarea/README.md [hunk 3]
```diff
@@ -0,0 +1,36 @@
...
+
+## Supported props
+
+Property name    | Type            | Default       | Description                    
+-----------------|-----------------|:-------------:|--------------------------------
+className        | string          | n/a           | Additional class name passed to wrapper
+disabled         | bool            | n/a           | Read-only textarea
+maxLength        | number          | n/a           | Define max number of characters in texarea
+placeholder      | string          | n/a           | Custom placeholder to show in textarea
+resize           | bool            | `true`        | Allow to resize textarea
```
Comment (rangoo94) [inline packages/textarea/README.md]: Actually this `resize` is not required, as it's just a single CSS property. But anyway, we may keep it.

File: packages/textarea/package.json [hunk 4]
```diff
@@ -0,0 +1,39 @@
...
+    "react": "^16.2.0",
+    "react-dom": "^16.2.0",
+    "react-textarea-autosize":"^6.1.0",
+    "@talixo/shared": "^0.1.0"
+  },
+  "peerDependencies": {
+    "prop-types": "^15.6.1",
+    "react": "^16.2.0",
+    "react-dom": "^16.2.0",
+    "react-textarea-autosize":"^6.1.0",
```
Comment (rangoo94) [inline packages/textarea/package.json]: `react-textarea-autosize` is not a peer dependency - it's not required at all.

File: packages/textarea/tests/Textarea.test.js [hunk 5]
```diff
@@ -0,0 +1,44 @@
...
+  it('renders textarea correctly', () => {
+    const wrapper = shallow(<Textarea />)
+
+    expect(wrapper).toMatchSnapshot()
+  })
+
+  it('renders as textarea correctly', () => {
+    const wrapper = shallow(<Textarea />)
+
+    expect(wrapper.html()).toMatch(/<textarea/)
```
Comment (rangoo94) [inline packages/textarea/tests/Textarea.test.js]: Why don't check if object **is** a textarea?

File: packages/textarea/src/Textarea.js [hunk 6]
```diff
@@ -0,0 +1,59 @@
...
+ * @param {object} props
+ * @param {string} [props.className]
+ * @param {bool} [props.disabled]
+ * @param {number} [props.maxLength]
+ * @param {string} [props.placeholder]
+ * @param {bool} [props.resize]
+ * @param {string|node} [props.TextareaComponent]
+ * @returns {React.Element}
+ */
+function Textarea (props) {
```
Comment (rangoo94) [inline packages/textarea/src/Textarea.js]: It is not handling self-control.

**Pull request merged**